### ISSUE TITLE
How to modify GeometricFilter if the rotation(not pose) between two images is known?

### ISSUE THREAD

**Issue by happyfrank** [openMVG/openMVG#1641]
For now we can get the rotation between two spherical images from camera imu. 
How to modify GeometricFilter if the rotation(not pose) between two images is known?
![1](https://user-images.githubusercontent.com/12987532/69204334-9bc09e00-0b81-11ea-9e5e-8fddee5ea168.jpg)
![5](https://user-images.githubusercontent.com/12987532/69204339-9f542500-0b81-11ea-893f-1491eaf95fc2.jpg)
![1](https://user-images.githubusercontent.com/12987532/69204795-ec84c680-0b82-11ea-8bb6-9539d8adc7ea.png)

**Comment by pmoulon** [comment]
Hello @happyfrank 
Thank you for your feedback.

OpenMVG is having a solver to deal with known rotation, but it is not yet included as part of the geometric filtering process. See here https://github.com/openMVG/openMVG/blob/develop/src/openMVG/multiview/solver_translation_knownRotation_kernel.hpp#L21

IMU information is often noisy.
Do you have any idea about the accuracy of your IMU, did you made an experiment to see its accuracy?

- Since you are taking multiple spherical images I assume you want to do either virtual tour or 360 reconstruction of museum.

I would advise to take more images and then choose the one you want to use at the end.
Or you can try this for a faster capture:
- capture your still (nice images)
- capture a video along the position of your captured stills (export the video as X frames per second (1 or 6))

Process the entire sequence, and then be happy to see everything localized ;-)

**Comment by happyfrank** [comment]
@pmoulon That's fantastic. I will try it. Thank you very much. ^v^

**Comment by pmoulon** [comment]
So to make the test you will have to compute the relative rotation from the two global rotation of your IMU.
See here how to compute it: https://github.com/openMVG/openMVG/blob/develop/src/openMVG/multiview/essential.cpp#L61

**Comment by pmoulon** [comment]
Which camera are you using (Theta, Insta)?

**Comment by happyfrank** [comment]
It's a 360 camera made by ourselves.

**Comment by happyfrank** [comment]
@pmoulon Hi, I try it. but the result was not right. 
Do you think it's possiable I can filter these wrong matches(more) and save correct matches(less) if rotation is known.
![1670717926](https://user-images.githubusercontent.com/12987532/69687712-df298800-10fe-11ea-8f78-ac651945ee2c.jpg)
Please forget my painting. ^v^

**Comment by pmoulon** [comment]
I see two ways to deal with such issue:
- Discard matches that are on top and bottom on the image (since you can assume you have distortions)
- Are those matches discarded if you have the rotations, I cannot say without testing with data, since we must see if the rotation you have from your IMU are accurate enough.
If you want some help from the community, feel free to ask by providing your two image and rotation matrix here or in private here openmvg-team[AT]googlegroups.com

**Comment by happyfrank** [comment]
@pmoulon That's all right. This is our [test_data.zip](https://github.com/openMVG/openMVG/files/3899338/test_data.zip). Thank you very much.

**Comment by happyfrank** [comment]
https://github.com/openMVG/openMVG/blob/2bf6e0822229c17494a1ef2277e08fa5bed7f986/src/openMVG/matching_image_collection/E_ACRobust_Angular.hpp#L111
By the way, is this **upper_bound_precision** should be setting Square? Or don't setting for better result?
